Common/Securities/Option/IQLRiskFreeRateEstimator.cs

```diff
@@ -0,0 +1,42 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QLNet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Securities.Option
+{
+    /// <summary>
+    /// Defines QuantLib risk free rate estimator for option pricing model. 
+    /// </summary>
+    interface IQLRiskFreeRateEstimator
```

Comment:
Those three interfaces and constant implementations are private by design. We would like user to customize estimators, but we dont want QL to leak into users code. Later I am adding more estimators and make them available to enduser, and introduce dsl that allows users estimate term structures w/o leaking QL.